src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,14 +1,33 @@
 package core.basesyntax.service;
 
 import core.basesyntax.dao.StorageDao;
-import core.basesyntax.dao.StorageDaoImpl;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private final StorageDao storageDao;
+
+    public RegistrationServiceImpl(StorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
 
     @Override
     public User register(User user) {
-        return null;
+        validateUser(user);
+        return storageDao.add(user);
+    }
+
+    private void validateUser(User user) {
+        if (user.getLogin() == null || user.getLogin().length() < 6) {
+            throw new InvalidUserException("Invalid login - min 6 characters");
+        }
+        if (user.getPassword() == null || user.getPassword().length() < 6) {
+            throw new InvalidUserException("Invalid password - min 6 characters");
+        }
+        if (user.getAge() == null || user.getAge() < 18) {
+            throw new InvalidUserException("Invalid age - min 18 years old");
```

Comment:
![image](https://github.com/mate-academy/jv-registration-validation/assets/20326858/0ac1c118-2f76-4571-a89f-d05136ed0dc1)

Reply:
not fixed